python/tvm/meta_schedule/task_scheduler/gradient_based.py

```diff
@@ -0,0 +1,146 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Gradient Based Task Scheduler"""
+import math
+
+from typing import TYPE_CHECKING, List, Optional
+from tvm._ffi import register_object
+from tvm._ffi.registry import register_func
+
+from tvm.ir import IRModule
+from ..measure_callback import MeasureCallback
+from ..builder import Builder
+from ..runner import Runner
+from ..database import Database
+from ..cost_model import CostModel
+from .task_scheduler import TaskScheduler
+
+from .. import _ffi_api
+
+if TYPE_CHECKING:
+    from ..tune_context import TuneContext
+
+
+@register_func("meta_schedule.task_scheduler.derive_similarity_tag")
+def derive_similarity_tag(mod: IRModule, log_base: float = 1.618) -> str:
+    """Get the tags for smilarity group creation
+
+    Parameters
+    ---------
+    mod : IRModule
+        The input workload.
+    log_base : float
+        The log base to normalize the flop count.
```

Comment:
Out of curiosity, why log base matters? It seems to me that you only use it for FLOP (to avoid overflow)?
```suggestion
        The log base to normalize the flop count. Default natural (1.618).
```

Reply:
Basically it won't matter much, the usage of base is only for workload grouping and then optimal flop estimation. This is same as the practice in autoscheduler.

Reply:
I see. Then it might be worthwhile to consider not exposing this option to users if we don't expect any benefit of other bases.

Reply:
Make sense. We allow the user to define the function and simply provide the function name so I think won't make a difference that we remove the log base.